src/main/webapp/app/instructor-course-dashboard/instructor-course-dashboard.component.ts

```diff
@@ -0,0 +1,113 @@
+    }
+
+    private loadCourse(courseId: number) {
+        this.courseService.findWithExercisesAndParticipations(courseId).subscribe(
+            (res: HttpResponse<Course>) => {
+                this.course = res.body;
+
+                let numberOfSubmissions = 0;
+                let numberOfAssessments = 0;
+
+                for (const exercise of this.course.exercises) {
+                    const validParticipations: Participation[] = exercise.participations.filter(participation => participation.initializationState === InitializationState.FINISHED);
+                    for (const participation of validParticipations) {
+                        for (const result of participation.results) {
+                            if (result.rated) {
+                                const tutorId = result.assessor.id;
+                                if (!this.tutorLeaderboardData[tutorId]) {
+                                    this.tutorLeaderboardData[tutorId] = {
+                                        tutor: result.assessor,
+                                        nrOfAssessments: 0
+                                    };
+                                }
+
+                                this.tutorLeaderboardData[tutorId].nrOfAssessments++;
+                            }
+                        }
+                    }
+
+                    exercise.participations = exercise.participations.filter(participation => participation.initializationState === InitializationState.FINISHED);
+                    exercise.numberOfAssessments = exercise.participations.filter(participation => participation.results.filter(result => result.rated).length > 0).length;
+
+                    numberOfAssessments += exercise.numberOfAssessments;
+                    numberOfSubmissions += exercise.participations.length;
+                }
+
+                this.stats.numberOfAssessments = numberOfAssessments;
+                this.stats.numberOfSubmissions = numberOfSubmissions;
+
+                this.dataNumbersForPieChart = [
+                    numberOfSubmissions - numberOfAssessments,
+                    numberOfAssessments
+                ];
+            },
+            (response: HttpErrorResponse) => this.onError(response.message)
+        );
+
+        this.courseService.getStatsForInstructors(courseId).subscribe(
+            (res: HttpResponse<StatsForInstructorDashboard>) => {
+                this.stats = Object.assign({}, this.stats, res.body);
+            },
+            (response: string) => this.onError(response)
+        );
+    }
+
+    calculatePercentage(numerator: number, denominator: number): number {
+        if (denominator === 0) {
+            return 0;
+        }
+
+        return Math.round( numerator / denominator * 100);
+    }
+
+    calculateClass(numberOfAssessments: number, length: number): string {
+        const percentage = this.calculatePercentage(numberOfAssessments, length);
+
+        if (percentage < 50) {
```

Comment:
maybe use constants for these values like 
```
export const MIN_POINTS_GREEN = 80;
export const MIN_POINTS_ORANGE = 40;
```

Reply:
I declared them as `readonly` properties of the class, since they are used only here